gradio/components.py

```diff
@@ -2761,6 +2761,138 @@ def style(self, *, full_width: Optional[bool] = None, **kwargs):
         return IOComponent.style(self, **kwargs)
 
 
+@document("change", "style")
+class UploadButton(Changeable, Uploadable, IOComponent, SimpleSerializable):
+    """
+    Used to create an upload button, when cicked allows a user to upload files that satisfy the specified file type or generic files (if file_type not set). The label is the text shown on the button, defaults to "Upload a File".
+    Preprocessing: passes the uploaded file as a {file-object} or {List[file-object]} depending on `file_count` (or a {bytes}/{List{bytes}} depending on `type`)
+    Postprocessing: expects function to return a {str} path to a file, or {List[str]} consisting of paths to files.
+    Examples-format: a {str} path to a local file that populates the component.
+    Demos:
+    """
+
+    def __init__(
+        self,
+        value: Optional[str | List[str] | Callable] = None,
+        *,
+        visible: bool = True,
+        elem_id: Optional[str] = None,
+        type: str = "file",
+        file_type: str = "file",
+        label: str = None,
+        **kwargs,
+    ):
+        """
+        Parameters:
+            value: Default file to display, given as str file path. If callable, the function will be called whenever the app loads to set the initial value of the component.
```

Comment:
I think the `value` parameter doesn't do anything right now, correct? It would be better if the `value` parameter is the text that is displayed on the button, similar to the `Button` component.

Reply:
Oh right, meant to remove it. I added a label param instead, but I guess that isn't consistent with `gr.Button`. Will change it.

Reply:
Actually using `value` won't work because the `postprocess` is called on the initial value and in this case the upload button postprocesses files. Am I right here?

Reply:
Do we even need a `postprocess()` function as `UploadButton` component can only be used as an input?

Reply:
Right, we actually don't need postprocess. Removed it

Reply:
And made it so that `value` is the button label